types/index.d.ts

```diff
@@ -15,7 +15,7 @@ declare function postgres<T extends PostgresTypeList>(options?: postgres.Options
  * @param options Connection options - default to the same as psql
  * @returns An utility function to make queries to the server
  */
-declare function postgres<T extends PostgresTypeList>(url: string, options?: postgres.Options<T>): postgres.Sql<PostgresTypeList extends T ? {} : { [type in keyof T]: T[type] extends {
+declare function postgres<T extends PostgresTypeList = {}>(url: string, options?: postgres.Options<T>): postgres.Sql<{ [type in keyof T]: T[type] extends {
```

Comment:
I am happy to change this, but I could not figure out why there was a condition for `PostgresTypeList extends T` and when I tried my fork it seemed to have the same effect when omitted.

Reply:
This is because sometimes TS infer `PostgresTypeList` (and not `{}`) under some conditions, then the `keyof T` is replace with `string` in the following object literal, and the `TTypes` generic become just a `Record<string, CustomTypeA | CustomTypeB | ...>`, which is quite useless. But maybe recent versions of typescript have fixed that, I checked again recently